tree/ntupleutil/v7/inc/ROOT/RNTupleImporter.hxx

```diff
@@ -16,41 +16,238 @@
 #ifndef ROOT7_RNTuplerImporter
 #define ROOT7_RNTuplerImporter
 
+#include <ROOT/REntry.hxx>
+#include <ROOT/RError.hxx>
+#include <ROOT/RField.hxx>
+#include <ROOT/RNTuple.hxx>
+#include <ROOT/RNTupleModel.hxx>
+#include <ROOT/RNTupleOptions.hxx>
 #include <ROOT/RStringView.hxx>
 
+#include <TFile.h>
+#include <TTree.h>
+
+#include <cstdlib>
+#include <map>
 #include <memory>
+#include <vector>
 
-class TTree;
+class TLeaf;
 
 namespace ROOT {
 namespace Experimental {
 
-namespace Detail {
-class RPageSink;
-}
-
 // clang-format off
 /**
 \class ROOT::Experimental::Detail::RNTupleImporter
 \ingroup NTuple
 \brief Converts a TTree into an RNTuple
 
-The class steers the conversion of a TTree into an RNTuple.
+Example usage:
+
+~~~ {.cpp}
+#include <ROOT/RNTupleImporter.hxx>
+using ROOT::Experimental::RNTupleImporter;
+
+auto importer = RNTupleImporter::Create("data.root", "TreeName", "output.root").Unwrap();
+// As required: importer->SetNTupleName(), importer->SetWriteOptions(), ...
+importer->Import().ThrowOnError();
+~~~
+
+The output file is created if it does not exist, otherwise the ntuple is added to the existing file.
+Note that input file and output file can be identical if the nutple is stored under a different name than the tree
+(use `SetNTupleName()`).
+
+By default, the RNTuple is compressed with zstd, independent of the input compression. The compression settings
+(and other output parameters) can be changed by `SetWriteOptions()`.
+
+Most RNTuple fields have a type identical to the corresponding TTree input branch. Exceptions are
```

Comment:
Should we mention here the current limitations (some are implicit from what RNTuple support but even when/if those are lifted in RNTuple they will need some additional work here). (eg.  Split objects, nested split collections, etc.)

Reply:
I added a [list of limitations](https://github.com/root-project/root/pull/11825/commits/7eff30014bd40b6f21dfd00a971f48d833c9db54), which is most likely not complete.

I'm not sure about split objects and nested split collections -- those should just work. In whatever way the object/collection is stored in TTree, it gets deserialized to a C++ type that is then picked up by RNTuple. RNTuple will not respect any "don't split" markers but the data should still be converted.

Reply:
> I'm not sure about split objects and nested split collections -- those should just work.

With my reading the code, I think it would not (the assert on line 137: `assert(firstLeaf);` might be fatal).   So we should re-check/test.  It is okay if it does not work yet for this PR.

Reply:
> RNTuple will not respect any "don't split" markers

When those are explicit from the user, it is probably a missing feature that should probably (eventually) be added.

Reply:
Oh, I see.  We need to further recurse into `GetListOfBranches()`. Ok, I extended the list of unsupported cases.

Reply:
But then... why does the [unit test with `CustomStructUtil`](https://github.com/root-project/root/pull/11825/files#diff-f1899eabcfcf04358e0796721cf3d3ddbe694445de7f8c566fc970c902f13e6eR353) work? Shouldn't it create sub branches for the members (default split level 1)?

Reply:
I was mis-remembering.  There is indeed a leaf in the top level branch in the case of split object.  So it actually should work.

Reply:
As discussed, I added a unit test with different split level settings and [a more complex custom object](https://github.com/root-project/root/pull/11825/commits/5dd331523a70362552340f2ddac113a9cd19afa7), including nested collections and inheritance.

Reply:
Thanks